src/encryptsave/mod.rs

```diff
@@ -0,0 +1,175 @@
+mod ll;
+pub mod errors;
+
+
+macro_rules! tox_res {
+    ( $res:ident <- $rexpr:expr, $err:ident, $r:expr ) => {
```

Comment:
Something like `{ something }.and_then(|r| tox_res!(err, { something(..., &mut err) ; r }))` (probably there's a better way to do it).